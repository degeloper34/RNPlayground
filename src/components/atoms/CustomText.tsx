import {Text, TextProps} from "../Themed";

export function CustomText(props: TextProps) {
  const decideFontFamilyByType = () => {
    switch (props.type) {
      case "regular":
        return "sora-regular";
      case "medium":
        return "sora-medium";
      case "bold":
        return "sora-bold";
      default:
        return "sora-regular";
    }
  };

  return (
    <Text
      {...props}
      allowFontScaling={false}
      style={[
        props.style,
        {
          fontFamily: decideFontFamilyByType(),
          fontSize: props.fontSize || 12,
          color: props.textColor,
          textDecorationLine: props.underline ? "underline" : "none",
        },
      ]}
    >
      {props.text}
    </Text>
  );
}
