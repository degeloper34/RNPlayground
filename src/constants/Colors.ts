export default {
  mainNight: "#21325E",
  mainNavy: "#3E497A",
  mainYellow: "#F1D00A",
  mainGrey: "#F0F0F0",
  white: "#fff",
  black: "#000",
};
