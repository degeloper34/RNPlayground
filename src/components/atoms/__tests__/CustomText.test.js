import renderer from "react-test-renderer";
import React from "react";
import {CustomText} from "../CustomText";
import Colors from "../../../constants/Colors";

describe("CustomText", () => {
  it("renders correctly with text", () => {
    const component = renderer
      .create(<CustomText text={"Hello, degeloper"} />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it("renders correctly with text and regular-type", () => {
    const component = renderer
      .create(<CustomText text={"Hello, degeloper"} type={"regular"} />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it("renders correctly with text and medium-type", () => {
    const component = renderer
      .create(<CustomText text={"Hello, degeloper"} type={"medium"} />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it("renders correctly with text and bold-type", () => {
    const component = renderer
      .create(<CustomText text={"Hello, degeloper"} type={"bold"} />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it("renders correctly with yellow text", () => {
    const component = renderer.create(
      <CustomText
        text={"Hello, degeloper"}
        type={"bold"}
        textColor={Colors.mainYellow}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
