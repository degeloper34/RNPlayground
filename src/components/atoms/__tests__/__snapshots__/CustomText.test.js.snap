// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomText renders correctly with text 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      undefined,
      Object {
        "color": "#21325E",
        "fontFamily": "sora-medium",
        "fontSize": 12,
        "textDecorationLine": "none",
      },
    ]
  }
>
  Hello, degeloper
</Text>
`;

exports[`CustomText renders correctly with text and bold-type 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      undefined,
      Object {
        "color": "#21325E",
        "fontFamily": "sora-bold",
        "fontSize": 12,
        "textDecorationLine": "none",
      },
    ]
  }
>
  Hello, degeloper
</Text>
`;

exports[`CustomText renders correctly with text and medium-type 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      undefined,
      Object {
        "color": "#21325E",
        "fontFamily": "sora-medium",
        "fontSize": 12,
        "textDecorationLine": "none",
      },
    ]
  }
>
  Hello, degeloper
</Text>
`;

exports[`CustomText renders correctly with text and regular-type 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      undefined,
      Object {
        "color": "#21325E",
        "fontFamily": "sora-regular",
        "fontSize": 12,
        "textDecorationLine": "none",
      },
    ]
  }
>
  Hello, degeloper
</Text>
`;

exports[`CustomText renders correctly with yellow text 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      undefined,
      Object {
        "color": "#F1D00A",
        "fontFamily": "sora-bold",
        "fontSize": 12,
        "textDecorationLine": "none",
      },
    ]
  }
>
  Hello, degeloper
</Text>
`;
